features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/jung/HierarchyLayoutAlgorithm.java

```diff
@@ -138,14 +138,19 @@ public int compare(Vertex o1, Vertex o2) {
         return jungGraph;
     }
 
-    private void applyLayoutPositions(final Collection<? extends Vertex> vertices, final edu.uci.ics.jung.algorithms.layout.Layout<VertexRef, Edge> layout, final Layout graphLayout) {
-        for(VertexRef v : vertices) {
+    private void applyLayoutPositions(final Collection<? extends Vertex> vertices, final HierarchyLayout<VertexRef, Edge> layout, final Layout graphLayout) {
+        final List<VertexRef> displayVertices = vertices.stream()
+                .map(v -> (VertexRef)v)
+                .collect(Collectors.toList());
+        layout.horizontalSqueeze(displayVertices);
```

Comment:
I did it this way since the squeeze is only performed on the displayed vertices, whereas the `HierarchyLayout` is calculated for the whole graph. We could also pass the set of displayed vertices in the `HierarchyLayout`'s constructor, but this looked even more convoluted.